ecmascript/parser/src/parser/input.rs

```diff
@@ -118,9 +118,13 @@ impl<I: Tokens> Iterator for Capturing<I> {
         match next {
             Some(ts) => {
                 let mut v = self.captured.borrow_mut();
-                if let Some(last) = v.last() {
+
+                // remove tokens that could change due to backtracing
+                while let Some(last) = v.last() {
                     if last.span.lo >= ts.span.lo {
-                        return Some(ts);
+                        v.pop();
```

Comment:
I'm not sure whether this is faster or removing a slice from the vector (which I'm not sure how to do in Rust).

Reply:
I'm not sure, but I think compiler will optimize it anyway